test_util/integration_test.py

```diff
@@ -69,6 +69,74 @@ def auth_cluster(cluster):
     return cluster
 
 
+@pytest.fixture(scope='module')
+def registry_cluster(cluster, request):
+    """Provides a cluster that has a registry deployed via marathon.
+    Note: cluster nodes must have hard-coded certs from dcos.git installed
+    """
+    if cluster.registry:
+        return cluster
+    registry_app = {
+            "id": "/registry",
+            "cmd": "docker run -p $PORT0:5000 mesosphere/test_registry:latest",
+            "cpus": 0.1,
+            "mem": 128,
+            "disk": 0,
+            "instances": 1,
+            "healthChecks": [{
+                "protocol": "COMMAND",
+                "command": {
+                    "value": "curl -sSfv https://registry.marathon.mesos.thisdcos.directory:$PORT0/v2/_catalog"},
+                "gracePeriodSeconds": 300,
+                "intervalSeconds": 60,
+                "timeoutSeconds": 20,
+                "maxConsecutiveFailures": 3,
+                "ignoreHttp1xx": True}],
+            "ports": [0],
+            "requirePorts": True
+            }
+    endpoints = cluster.deploy_marathon_app(registry_app)
+    cluster.registry = 'registry.marathon.mesos.thisdcos.directory:'+str(endpoints[0].port)
+    with open('/test_server.py', 'r') as fh:
+        test_server = fh.read()
+    with open('/test_server_Dockerfile', 'r') as fh:
+        dockerfile = fh.read()
+    docker_cmds = """
+#!/bin/bash
+mkdir tmp
+cat <<EOF > tmp/test_server.py
+{test_server}
+EOF
+cat <<EOF > tmp/Dockerfile
+{dockerfile}
+EOF
+docker build -t {registry}/test_server tmp/
+docker push {registry}/test_server
+sleep 36000
+""".format(registry=cluster.registry, test_server=test_server, dockerfile=dockerfile)
+    docker_build_and_push_app = {
```

Comment:
I'd like to see us put this in dockerhub too, and pin to a version in the test to make it easy to update the test against the hosted image in DH.